docs/source/posts/2023/2023-08-24-final-report-tvcastillod.rst

```diff
@@ -0,0 +1,175 @@
+
+ODF actor implemented with SDF
+******************************
+
+HARDI-based techniques require more images than DTI, however, they model the diffusion directions as probability distribution functions (PDFs), and the fitted values are returned in the form of orientation distribution functions (ODFs). ODFs are more diffusion sensitive than the diffusion tensor and, therefore, can determine the structure of multi-directional voxels very common in the white matter regions of the brain [5]_. The current actor to display this kind of glyphs is the ``odf_slicer`` which given an array of spherical harmonics (SH) coefficients render a grid of ODFs, which are created from a sphere with a specific number of vertices that fit the data.
+
+For the application of this model using the same SDF raymarching techniques, we need the data of the SH coefficients, which are used to calculate the orientation distribution function (ODF) described `here <https://dipy.org/documentation/1.7.0/theory/sh_basis/>`_. There are different SH bases that can be used, but for this first approach we focus on ``descoteaux07`` (as labeled in dipy). After performing the necessary calculations, we obtain an approximate result to the current implementation of FURY as seen below.
+
+.. image:: https://user-images.githubusercontent.com/31288525/260909561-fd90033c-018a-465b-bd16-3586bb31ca36.png
+    :width: 580
+    :align: center
+
+Already with a first implementation we start to solve some issues related to direction, color and data handling, in order to obtain exactly the same results of the current implementation.
+
+Details on the issues:
+
+- *The direction and the scaling*: When the shape of the ODF is more sphere-like the size of the glyph is smaller, so right now I had to adjust it manually but the idea is to find a relationship between the coefficients and the final object size so it can be automatically scaled. Additionally, as seen in the image the direction does not match, for this you can make an adjustment in the calculation of the spherical coordinates, or pass the direction information directly.
+- *Pass the coefficients data efficiently*: Right now I'm creating one actor per glyph since I'm using a *uniform* array to pass the coefficients, but the idea is to pass all the data at once. A first idea is to encode the coefficients data through a texture and retrieve them in the fragment shader.
```

Comment:
- `Right now I'm` -> `I'm currently`
- `at once` -> `simultaneously`
- `A first` -> `The first`